src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -98,11 +108,24 @@ private function createNamedSerializerTags(ContainerBuilder $container, string $
         }
     }
 
-    private function bindDefaultContext(ContainerBuilder $container, array $services, array $defaultContext): void
+    private function bindDefaultContext(ContainerBuilder $container, array $services, array $defaultContext, ?string $circularReferenceHandler, ?string $maxDepthHandler): void
     {
         foreach ($services as $id) {
             $definition = $container->getDefinition((string) $id);
-            $definition->setBindings(['array $defaultContext' => new BoundArgument($defaultContext, false)] + $definition->getBindings());
+
+            $context = $defaultContext;
+            if (str_starts_with((string) $id, 'serializer.normalizer.object')) {
```

Comment:
Shouldn't we test instead the FQCN of the service?
I mean, here it sounds like we introduce a convention between the service ID and the class inheritance.

Reply:
The class for the child services is not set at this moment. What I could do is set it manually when the child services are registered.